test/checkChallenge.test.js

```diff
@@ -0,0 +1,1627 @@
+
+      config.minperiodtimecorr = 14400;
+      config.maxperiodtimecorr = 259200;
+      
+      const result = conn.checkChallengeMinMax(notification);
+      
+      assert.deepEqual(result, ({ reject: false }));
+    });
+
+    it('accept period time correspondence edge max', () => {
+      const notification = base_challenge({ ranked: false, time_control: { system: "byoyomi", time_control: "byoyomi", speed: "correspondence", main_time: 1, periods: 1, period_time: 259200 } });
+
+      config.minperiodtimecorr = 14400;
+      config.maxperiodtimecorr = 259200;
+      
+      const result = conn.checkChallengeMinMax(notification);
+      
+      assert.deepEqual(result, ({ reject: false }));
+    });
+
+    it('accept period time correspondence too high', () => {
+      const notification = base_challenge({ ranked: false, time_control: { system: "byoyomi", time_control: "byoyomi", speed: "correspondence", main_time: 1, periods: 1, period_time: 512000 } });
+
+      config.minperiodtimecorr = 14400;
+      config.maxperiodtimecorr = 259200;
+      
+      const result = conn.checkChallengeMinMax(notification);
+      
+      assert.deepEqual(result, ({ reject: true,   msg: 'Maximum Period Time for correspondence games in byoyomi is 3 days, please reduce Period Time.' }));
+    });
+
+  });
+
+  describe('Some time controls have no main time, do not check maintime', () => {
+
+    it('simple timecontrol accepts any main time, even if too low', () => {
+      const notification = base_challenge({ ranked: false, time_control: { system: "simple", time_control: "simple", speed: "live", per_move: 1 } });
+
+      config.minmaintimelive = 60;
+      config.maxmaintimelive = 300;
+      
+      const result = conn.checkChallengeMinMax(notification);
+      
+      assert.deepEqual(result, ({ reject: false }));
+    });
+
+    it('simple timecontrol accepts any main time, even if between min and max', () => {
+      const notification = base_challenge({ ranked: false, time_control: { system: "simple", time_control: "simple", speed: "live", per_move: 180 } });
+
+      config.minmaintimelive = 60;
+      config.maxmaintimelive = 300;
+      
+      const result = conn.checkChallengeMinMax(notification);
+      
+      assert.deepEqual(result, ({ reject: false }));
+    });
+
+    it('simple timecontrol accepts any main time, even if too high', () => {
+      const notification = base_challenge({ ranked: false, time_control: { system: "simple", time_control: "simple", speed: "live", per_move: 9999 } });
+
+      config.minmaintimelive = 60;
+      config.maxmaintimelive = 300;
+      
+      const result = conn.checkChallengeMinMax(notification);
+      
+      assert.deepEqual(result, ({ reject: false }));
+    });
+    
+  });
+
+  describe('Non-Byoyomi time controls have no period time, do not check periods', () => {
```

Comment:
no, if you check the time_control notification samples i provided, you can see only byoyomi has a `notification.time_control.periods` and this is the thing we test in periods

if we want to add other time settings that work in a similar way than periods, we could either add them in the same periods number check (which i wouldnt like because they seem to function quite differently), or make other time settings options like `--minnumberofstonesperperiod` `--maxnumberofstonesperperiod` `--minmaxtime` `--maxmaxtime`, etc :)

but for this pr i just wanted to replicate our old functionning, so if this looks like a good idea to you to add these options or change what our code currently does, we should make it in another pr i think or it would change too much things at once no?